ports/mimxrt10xx/common-hal/busio/UART.c

```diff
@@ -198,15 +198,18 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
       claim_pin(self->tx_pin->pin);
 
     if (self->rx_pin != NULL) {
-        ringbuf_alloc(&self->rbuf, receiver_buffer_size, true);
+        // The LPUART ring buffer wastes one byte to distinguish between full and empty.
+        self->ringbuf = gc_alloc(receiver_buffer_size + 1, false, true /*long-lived*/);
```

Comment:
Because LPUART provides its own ring buffer implementation. It doesn't need a ring buffer, it just needs a buffer. So all the code was doing is a `gc_alloc()` on `rbuf.buf`. The ringbuf operations in `ringbuf.h` were never called.

Reply:
@jepler does this make sense to you?